WalletWasabi/Tor/TorSettings.cs

```diff
@@ -62,13 +74,24 @@ public static string GetTorBinaryFilePath(OSPlatform? platform = null)
 
 		public string GetCmdArguments(EndPoint torSocks5EndPoint)
 		{
-			return string.Join(
-				" ",
+			List<string> arguments = new()
+			{
 				$"--SOCKSPort {torSocks5EndPoint}",
+				$"--CookieAuthentication 1",
+				$"--ControlPort {ControlEndpoint.Port}",
+				$"--CookieAuthFile \"{CookieAuthFilePath}\"",
 				$"--DataDirectory \"{TorDataDir}\"",
 				$"--GeoIPFile \"{GeoIpPath}\"",
 				$"--GeoIPv6File \"{GeoIp6Path}\"",
-				$"--Log \"notice file {LogFilePath}\"");
+				$"--Log \"notice file {LogFilePath}\""
+			};
+
+			if (TerminateOnExit && OwningProcessId is not null)
```

Comment:
Why is the addition of `OwningProcessId` depends on `TerminateOnExit`? Isn't it useful anyway, like monitoring?

Reply:
So if you remove `TerminateOnExit`, Tor would actually terminate when WW terminates because Tor would notice that WW process ID does not exist. That's the purpose it was used here.

Reply:
I got it. In that case, it has to be there.